kv/dist_sender.go

```diff
@@ -600,10 +618,48 @@ func (ds *DistSender) sendAttempt(trace *tracer.Trace, args proto.Request, desc
 // This may temporarily adjust the request headers, so the proto.Call
 // must not be used concurrently until Send has returned.
 func (ds *DistSender) Send(ctx context.Context, call proto.Call) {
-	args := call.Args
+	// TODO(tschottdorf): provisional code that wraps everything in a one-
```

Comment:
I'll work on top of that (in this PR) until it reaches a mergable state. But since the thing is only going to grow, I want to get eyes on each individual stepping stone.